### PR TITLE
libjpeg-turbo: Use spack flags for cmake building with Fujitsu compiler.

### PR DESCRIPTION
**Pull request by t-karatsu**
Cmake executed try_compile when installing libjpeg-turbo, but it was failed with Fujitsu compiler.
Because cmake didn't use compiler flags and Fujitsu compiler needs flags for each environment.

So, we will add spack cflags to cmake options only when using Fujitsu compiler.
※ Libjpeg-turbo doesn't need CXXFLAGS or FFLAGS.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libjpeg-turbo/package.py [hunk 1]
```diff
@@ -48,6 +48,13 @@ def install(self, spec, prefix):
     @when('@1.5.90:')
     def install(self, spec, prefix):
         cmake_args = ['-GUnix Makefiles']
+
+        # For Fujitsu compiler(fj)
+        # Use spack compiler flags for cmake.
+        if self.compiler.name == 'fj':
+            cmake_args.extend(['-DCMAKE_C_FLAGS={0}'
+                              .format(env["SPACK_CFLAGS"])])
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: @tkameyama can you set this as 

```
'-DCMAKE_C_FLAGS={0}'
.format(spec.compiler_flags['cflags'])
```

This is a slightly-more robust way to access the flags. Ideally we could use some of the built-in flag-handling logic for `CMakePackage` but it appears that this uses different build systems depending on the version.

Comment (t-karatsu) [reply var/spack/repos/builtin/packages/libjpeg-turbo/package.py]: Thanks for your advice!
I changed way to take compiler flags on commit of `libjpeg-turbo: fix way to take compiler flags`. 

`spec.compiler_flags['cflags']` was list, so I used phrase `.join`.

**Comment by scheibelp** [comment]
I thought about this more and have an idea for how to make this more generic: I think it would be good to support CMake compiler flags for cases in addition to Fujitsu.

If you have time, can you try it out? It is based on the flag_handler functionality described at https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags. This also avoids duplicating cflags: currently the flags are passed to -DCMAKE_C_FLAGS and also to the compiler wrapper by Spack. 

```
def flag_handler(self, name, flags):
    if self.spec.satisfies('@1.5.90:'):
        return (None, None, flags)
    else:
        # compiler flags for earlier version are injected into the
        # spack compiler wrapper
        return (flags, None, None)

def flags_to_build_system_args(self, flags):
    # This only handles cflags, other flags are discarded
    cmake_flag_args = []
    if 'cflags' in flags and flags['cflags']:
        cmake_flag_args.append('-DCMAKE_C_FLAGS={0}'.format(
            ' '.join(flags['cflags']))
    self.cmake_flag_args = cmake_flag_args

@when('@1.5.90:')
def install(self, spec, prefix):
    cmake_args = ['-GUnix Makefiles']
    if hasattr('cmake_flag_args'):
        cmake_args.extend(self.cmake_flag_args)
    cmake_args.extend(std_cmake_args) # as before
    ...
```

**Comment by t-karatsu** [comment]
OK. Our team will consider fixes about it. Please give me some time to consider it.

**Comment by t-karatsu** [comment]
I implemented the source code with some modifications to your changes, and Installing `libjpeg-turbo` was succeeded! I pushed changes on this pullreq. Could you check  commit of "libjpeg-turbo: Use spack flags for cmake when building with Fujitsu compiler."?

**Comment by scheibelp** [comment]
Thanks! This looks great.

**Comment by t-karatsu** [comment]
Thanks for the merge!

**Pull request merged**